webhook/main.go

```diff
@@ -28,23 +33,59 @@ func main() {
 		redisPassword = "" // Default password (empty in this case)
 	}
 
-	client := redis.NewClient(&redis.Options{
-		Addr:     redisAddress,
-		Password: redisPassword,
-		DB:       0, // Default DB
-	})
+	client, err := createRedisClient()
+	if err != nil {
+		log.Fatalf("Failed to create Redis client: %v", err)
+	}
+	log.Println(redisPassword, redisAddress)
+
 	// Create a channel to act as the queue
 	webhookQueue := make(chan redisClient.WebhookPayload, 100) // Buffer size 100
 
 	go queue.ProcessWebhooks(ctx, webhookQueue)
 
 	// Subscribe to the "transactions" channel
-	err := redisClient.Subscribe(ctx, client, webhookQueue)
+	err = redisClient.Subscribe(ctx, client, webhookQueue)
 
 	if err != nil {
 		logging.WebhookLogger(logging.ErrorType, fmt.Errorf("error initializing connection: %s", err))
+		log.Fatalf("error initializing connection: %v", err)
+		return
 	}
 
 	select {}
+}
+
+func createRedisClient() (*redis.Client, error) {
```

Comment:
We must allow optional parameters here. We might need to instantiate Redis with a different configuration than the one in the environment

Reply:
What can be the optional parameters ?

Reply:
Host password and bd